src/core/xlio.h

```diff
@@ -36,6 +36,7 @@
 #include <signal.h>
 #include <sys/poll.h>
 
+#ifdef __cplusplus
```

Comment:
This makes the whole API unavailable to C. What you want instead is this:

#ifdef __cplusplus
extern "C" {
#endif

And similarly below for the closing bracket.

Reply:
OK